Lesson16/main.js

```diff
@@ -90,15 +91,53 @@ async function configUIfromFetchData() {
   }
 }
 
+function addNewIcon(date, element) {
+  const today = format(new Date(), "yyyy,MM,dd");
+  const articleDate = format(new Date(date), "yyyy,MM,dd");
+  const periodOfDays = differenceInCalendarDays(
+    new Date(today),
+    new Date(articleDate)
+  );
+  const newArrivalDays = 3;
+  if (periodOfDays <= newArrivalDays) {
+    const newIcon = document.createElement("img");
+    newIcon.src = "./new-icon.svg";
+    newIcon.classList.add("new-icon");
+    element.appendChild(newIcon);
+  }
+}
```

Comment:
This function has two different processes, so I thought it would be better to separate the functions.
IMO
```js
function checkSpecifiedPeriod(date) {
  const newArrivalDays = 3;
  const today = format(new Date(), "yyyy,MM,dd");
  const articleDate = format(new Date(date), "yyyy,MM,dd");
  const periodOfDays = differenceInCalendarDays(
    new Date(today),
    new Date(articleDate)
  );
  const result = periodOfDays <= newArrivalDays;
  return result;
}

const addNewIcon = (element) => {
  const newIcon = document.createElement("img");
  newIcon.src = "./new-icon.svg";
  newIcon.classList.add("new-icon");
  element.appendChild(newIcon);
};
```
and
```js
async function createArticleElements({ article }) {
  const ul = document.getElementById("js-tab-content__list");
  const frag = document.createDocumentFragment();
  for (let i = 0; i < article.length; i++) {
    const metaWrapper = document.createElement("div");
    metaWrapper.classList.add("meta-wrapper");
    if (checkSpecifiedPeriod(article[i].date)) {
      addNewIcon(metaWrapper);
    }
    addCommentCount(article[i], metaWrapper);

    const li = document.createElement("li");
    const anchor = document.createElement("a");
    anchor.href = "#";
    anchor.insertAdjacentHTML("beforeend", article[i].title);
    frag.appendChild(li).appendChild(anchor).after(metaWrapper);
  }
  ul.appendChild(frag);
}
```

Reply:
Thanks again for your review!
I learned a lot from the reference code.

For the function whether it is within 3 days or not.I named it `isSpecifiedPeriod`.
What do you think?
f07ce98

Reply:
> I named it isSpecifiedPeriod.
What do you think?

I think so good!